### PR TITLE
Improve CLI stuff and allow running on Windows

### PR DESCRIPTION
**Pull request by mlocati**
This is a big one :wink:

Here's what's included:

1. the php executable used to run the websocket server is configurable (default: `php`)
2. the debug log is configurable, and defaults to `/var/log/subscription_server.log` (on Windows, it defaults to `%TEMP%\subscription_server.log`)
3. I replaced the `Websocket` class that started the websocket servers with a standard concrete5 CLI command (see the new `WebsocketRunnerCommand` class)
4. I renamed the `WebsocketHelpers` class to `WebsocketService`, making all its methods non static and adding all the required dependencies to its `__construct`.
5. I patched the `WebsocketService` methods that were posix specific, so that they now should also work on Windows

### CODE REVIEW

File: src/Console/WebsocketRunnerCommand.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            $this->output->error('The port number must be an integer between 1 and 65535');
+
+            return 1;
+        }
+        $serverIP = $config->get('concrete5_graphql_websocket::websocket.server_ip');
+        $this->output->write("Checking availability of port {$port} on {$serverIP}... ");
+        $connection = @fsockopen($serverIP, $port);
+        if (is_resource($connection)) {
+            fclose($connection);
+            $this->output->error("no server found!\n");
```
Comment (marcokuoni) [inline src/Console/WebsocketRunnerCommand.php]: $this->output->error("server already exists on this port!\n");

File: controllers/single_page/dashboard/system/environment/graphql.php [hunk 2]
```diff
@@ -90,10 +90,11 @@ public function update_entity_settings()
                         $config->save('concrete5_graphql_websocket::websocket.debug', false);
                         $servers = (array) ($config->get('concrete5_graphql_websocket::websocket.servers'));
                         $config->save('concrete5_graphql_websocket::websocket.servers', []);
+                        $websocketService = $this->app->make(WebsocketService::class);
                         foreach ($servers as $port => $pid) {
```
Comment (marcokuoni) [inline controllers/single_page/dashboard/system/environment/graphql.php]: foreach ($servers as $pid) {

**Review by marcokuoni — COMMENTED** [review]
Looks really cool. Nice to know how to add stuff to CLI and how to support Window. But i think the line 33 in WebsocketRunnerCommand should be different. When you are on my track, pls change and commit, then i will merge everything. Thanks.

File: src/Console/WebsocketRunnerCommand.php (hunk 1 above)
Comment (mlocati) [reply src/Console/WebsocketRunnerCommand.php]: Whoops, you are right of course! Done.

File: controllers/single_page/dashboard/system/environment/graphql.php (hunk 2 above)
Comment (mlocati) [reply controllers/single_page/dashboard/system/environment/graphql.php]: I didn't touch this line ([was already like that](https://github.com/lemonbrain-mk/graphql_websocket/blob/1.2.5/controllers/single_page/dashboard/system/environment/graphql.php#L93)), but I agree: getting the `$port` is useless here. :+1:

**Review by marcokuoni — APPROVED** [review]
perfect

**Comment by marcokuoni** [comment]
There was also a typo in WebsocketService escapeshellarg i changed it for the next release

**Comment by marcokuoni** [comment]
I also added that the CLI commands get started in the background. Otherwise the apache can't start the websocketserver, cause the apache process hangs up. This new Version is in 1.2.6. Would be nice if you could check if this also works on a windows machine, i'm not sure if it is also a '&' at the end.

**Comment by marcokuoni** [comment]
I found an other bug for the !window version. there was a missing '>'. cause of this its now version 1.2.7

**Pull request merged**